Dirk Tilger <dirk@miriup.de>:
  For the mntent.patch: e2fsprogs is using a GNU libc specific
  operation: setmntent and endmntent. Both operate on /etc/mtab, which
  doesn't exist on MacOSX, thus they are simply not called on Darwin.

http://bugs.gentoo.org/show_bug.cgi?id=263841

--- misc/ismounted.c	2008-05-05 21:00:01 +0400
+++ misc/ismounted.c	2008-05-05 21:00:16 +0400
@@ -86,8 +86,10 @@
 	if ((f = fopen(mtab_file, "r")) == NULL)
 		return errno;
 
+	#ifndef __APPLE__
	if ((f = setmntent (mtab_file, "r")) == NULL)
		return errno;
+	#endif
 	if (stat(file, &st_buf) == 0) {
 		if (S_ISBLK(st_buf.st_mode)) {
 #ifndef __GNU__ /* The GNU hurd is broken with respect to stat devices */
@@ -177,7 +173,11 @@
 
 	retval = 0;
 errout:
+	#ifdef __APPLE__
+	fclose (f);
+	#else
	endmntent (f);
+	#endif
 	return retval;
 }

